airflow/hooks/abstract_db_hook.py

```diff
@@ -0,0 +1,65 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+class AbstractDbHook:
```

Comment:
`BaseDbHook` would be my preferred naming for this, as there is a specific concept in python of an Abstract Base Class (or ABC) and this is not one https://docs.python.org/3/library/abc.html

(I would be happy for it to _be_ one, but it isn't currently.)

Reply:
Right. not a good name. Will think about a better one (see further comment)